### DISCUSSION TITLE
How to filter json list with `contains` in MySQL?

### DISCUSSION THREAD

**Discussion by lz1998** [SeaQL/sea-orm#551] [Q&A]
MySQL Table
|  id   | setting  |
|  ----  | ----  |
| 1  | {"owners":["aaa","bbb","ccc"]} |
| 2  | {"owners":["ccc","ddd","eee"]} |

I want to find those records which contains "aaa" in "owners"

```rust
q.filter(Expr::expr(Expr::cust("setting->>'$.owners'")).contains("aaa"));
```

```bash
error[E0277]: the trait bound `sea_orm::sea_query::SimpleExpr: From<&str>` is not satisfied
    --> src/database/project.rs:92:74
     |
92   |     q = q.filter(Expr::expr(Expr::cust("setting->>'$.owners'")).contains(owner));
     |                                                                 -------- ^^^^^ the trait `From<&str>` is not implemented for `sea_orm::sea_query::SimpleExpr`
     |                                                                 |
     |                                                                 required by a bound introduced by this call
     |
     = help: the following implementations were found:
               <sea_orm::sea_query::SimpleExpr as From<Expr>>
     = note: required because of the requirements on the impl of `Into<sea_orm::sea_query::SimpleExpr>` for `&str`

```

**Comment by Ilqjx** [comment]
`xxx::Entity::find().filter(xxx::Column::Setting.contains('aaa')).all(&db).await`

https://www.sea-ql.org/SeaORM/docs/basic-crud/select#find-with-conditions-and-orders

**Reply by lz1998** [reply]
> `xxx::Entity::find().filter(xxx::Column::Setting.contains('aaa')).all(&db).await`
> 
> https://www.sea-ql.org/SeaORM/docs/basic-crud/select#find-with-conditions-and-orders

`setting` is JSON

**Comment by billy1624** [comment]
Hey @lz1998, replacing `contains` with `is_in`?
https://docs.rs/sea-query/0.22.0/sea_query/expr/struct.Expr.html#method.is_in

**Reply by lz1998** [reply]
> Hey @lz1998, replacing `contains` with `is_in`? https://docs.rs/sea-query/0.22.0/sea_query/expr/struct.Expr.html#method.is_in

I have tried `select*from project where 'aaa' in setting->"$.owners"`.

And an error occurs:
`[42000][1064] You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'setting->"$.owners"' at line 1`

**Reply by billy1624** [reply]
This is MySQL error. I guess you are running the SQL on older version of MySQL? Which does not support the JSON search syntax?
REF: https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html
